test/apps/iodemo/io_demo.cc

```diff
@@ -877,7 +920,11 @@ class DemoServer : public P2pDemoCommon {
 
             if (_status == UCS_OK) {
                 if (_server->opts().validate) {
-                    validate(*_iov, _sn);
+                    try {
+                        validate(*_iov, _sn);
+                    } catch (const std::exception &e) {
+                        handle_validate_error(e, _conn);
+                    }
```

Comment:
seems this pattern is repeating itself, maybe add helper function like validate which accepts connection pointer?

Reply:
thanks, it help me to find better solution